### PR TITLE
Implemented product service api for fetching products

### PR DESCRIPTION
**Pull request by akshaynr**
All tasks are completed

Additional:
Async/await is used in lambda functions
ES6 modules are used for Product Service implementation
Custom Webpack/ESBuild/etc is manually configured for Product Service. Not applicable for preconfigured/built-in bundlers that come with templates, plugins, etc.
SWAGGER documentation is created for Product Service
Lambda handlers are covered by basic UNIT tests (NO infrastructure logic is needed to be covered)
Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase.
Main error scenarios are handled by API ("Product not found" error). Also Invalid Product ID is taken care

Please check following urls: 
GET - https://0i8erly05h.execute-api.eu-west-1.amazonaws.com/products
GET - https://0i8erly05h.execute-api.eu-west-1.amazonaws.com/products/{productId}
GET - https://0i8erly05h.execute-api.eu-west-1.amazonaws.com/swagger
GET - https://0i8erly05h.execute-api.eu-west-1.amazonaws.com/swagger.json

### CODE REVIEW

File: product-service/serverless.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#package:
+#  patterns:
+#    - '!exclude-me.js'
+#    - '!exclude-me-dir/**'
+#    - include-me.js
+#    - include-me-dir/**
+
+custom:
+  autoswagger:
+    apiType: 'http'
```
Comment (OMManko) [inline product-service/serverless.yml]: could you check links for swagger? config is correct but you get different links and testing endpoints in swagger fails
image

File: product-service/handlers/getProductsById.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        'Access-Control-Allow-Credentials': true,
+    };
+
+    const productsData = await mockAsyncCall(productsMockData);
+    let productId = null;
+
+    if(event && event.pathParameters && event.pathParameters.productId != null){
+        productId = event.pathParameters.productId;
+    } else{
+        return {
```
Comment (OMManko) [inline product-service/handlers/getProductsById.js]: You can create some util function to return response passing proper params. this way you will avoid code duplication and won't create headers for every function
It is applicable for both for products and productId, success and error responses

File: product-service/handlers/getProductsById.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            statusCode: 400,
```
Comment (OMManko) [inline product-service/handlers/getProductsById.js]: I would suggest to move status codes to constant file

File: product-service/serverless.yml (hunk 1 above)
Comment (akshaynr) [reply product-service/serverless.yml]: Dev stage was not creating for api. Hence removed swagger custom settings

File: product-service/handlers/getProductsById.js (hunk 2 above)
Comment (akshaynr) [reply product-service/handlers/getProductsById.js]: Created separate function as suggested

File: product-service/handlers/getProductsById.js (hunk 3 above)
Comment (akshaynr) [reply product-service/handlers/getProductsById.js]: Created constant file as requested

**Pull request merged**